src/Meilisearch/Filter/CompositeFilterBuilder.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\Meilisearch\Filter;
+
+use Setono\CompositeCompilerPass\CompositeService;
+
+/** @extends CompositeService<FilterBuilderInterface> */
+final class CompositeFilterBuilder extends CompositeService implements FilterBuilderInterface
+{
+    public function build(array $facets, array $facetsValues): array
+    {
+        $filters = [];
+
+        foreach ($this->services as $filterBuilder) {
+            $filters = array_merge($filters, $filterBuilder->build($facets, $facetsValues));
```

Comment:
I think the PHP plugin in PHPStorm warns against this. You can read about it here: https://www.exakat.io/en/speeding-up-array_merge/

It should be `$filters[] = $filterBuilder->build($facets, $facetsValues)`

and then outside the loop:

`$filters = array_merge(...$filters);`